src/Microsoft.OData.Client/IHttpClientProvider.cs

```diff
@@ -9,21 +9,19 @@
 namespace Microsoft.OData.Client
 {
     /// <summary>
-    /// Provides the <see cref="HttpClientHandler"/> instance(s) used by OData Client
+    /// Provides the <see cref="HttpClient"/> instance(s) used by OData Client
     /// to execute requests.
     /// </summary>
-    public interface IHttpClientHandlerProvider
+    public interface IHttpClientProvider
     {
         /// <summary>
-        /// Returns an <see cref="HttpClientHandler"/> instance that OData Client should
+        /// Returns an <see cref="HttpClient"/> instance that OData Client should
         /// used to execute a request. This method is called per request.
         /// </summary>
-        /// <returns>The <see cref="HttpClientHandler"/> instance that will be used to execute the request.</returns>
+        /// <returns>The <see cref="HttpClient"/> instance that will be used to execute the request.</returns>
         /// <remarks>
-        /// It's possible that the returned <see cref="HttpClientHandler"/> may be mutated inside a request hook
-        /// e.g. via <see cref="HttpClientRequestMessage.Credentials"/>.
-        /// OData Client does not dispose the <see cref="HttpClientHandler"/> instance.
+        /// OData Client does not dispose the <see cref="HttpClient"/> instance.
         /// </remarks>
-        HttpClientHandler GetHttpClientHandler();
+        HttpClient GetHttpClient();

```

Comment:
Should this method accept arguments? Do you think there's a use case for creating different `HttpClient` instances based on some request context information, like the request URI?